controllers/capability/reconiler_test.go

```diff
@@ -216,14 +216,20 @@ func TestReconcile_DeleteStatefulSetIfOldLabelsAreUsed(t *testing.T) {
 	assert.False(t, deleted)
 
 	r.Instance.Spec.Proxy = &dynatracev1alpha1.DynaKubeProxy{Value: testValue}
-	labels := make(map[string]string)
-	labels[OldKeyActiveGate] = "dynakube"
-	r.Instance.Labels = labels
 	desiredSts, err = r.buildDesiredStatefulSet()
 	require.NoError(t, err)
+	correctLabels := make(map[string]string)
+	for key, value := range desiredSts.Labels {
+		correctLabels[key] = value
+	}
+	delete(desiredSts.Labels, KeyActiveGate)
+	delete(desiredSts.Labels, KeyDynatrace)
+	delete(desiredSts.Labels, KeyFeature)
+	desiredSts.Labels["activegate"] = "dynakube"
```

Comment:
If the intention is just to replace the whole map, I may suggest doing,

```go
correctLabels := desiredSts.Labels
desiredSts.Labels = map[string]string { "activegate": "dynakube" }
```

Reply:
Yes, that's way better